### ISSUE TITLE
Unable to build 'AnyCpu' executable in .net core

### ISSUE THREAD

**Issue by ChristophHornung** [dotnet/sdk#13329]
With .net Framework 4.x it was possible to build an executable as AnyCpu. It would start as 32bit on a 32bit machine and 64bit on a 64bit machine.

With the simple .net core [project](https://github.com/dotnet/sdk/files/5162765/Env32Test.zip) attached if I build the .exe

- on a 32bit system: It will start on both 32bit and 64bit but both times in 32bit mode
- on a 64bit system: It will only start on 64bit, on 32bit it displays ("The version of this file is not compatible with the version of Windows you're running.")

If I start via `dotnet Env32Test.dll` though it runs on the same bit-value as the machine.

Is it possible to create the same kind of executable as before?

**Comment by huoyaoyuan** [comment]
No, since the beginning of .Net Core. "AnyCPU executable" was specially treated by Windows, but .Net Core isn't specifically known by Windows.
